### PR TITLE
[MKC] Implement Watcher of Hours

### PR DESCRIPTION
**Pull request by PurpleCrowbar**

### CODE REVIEW

**Comment by PurpleCrowbar** [comment]
[[Watcher of Hours]]

**Comment by PurpleCrowbar** [comment]
Verify failure unrelated to this PR

File: Mage/src/main/java/mage/abilities/common/CounterRemovedFromSourceWhileExiledTriggeredAbility.java [hunk 1]
```diff
@@ -42,6 +51,8 @@ public boolean checkEventType(GameEvent event, Game game) {
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
-        return event.getData().equals(counterType.getName()) && event.getTargetId().equals(this.getSourceId());
+        return event.getData().equals(counterType.getName())
+                && event.getTargetId().equals(this.getSourceId())
+                && (!onlyController || event.getPlayerId() == this.getControllerId());
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/common/CounterRemovedFromSourceWhileExiledTriggeredAbility.java:56]: compare UUID with equals

Comment (PurpleCrowbar) [reply Mage/src/main/java/mage/abilities/common/CounterRemovedFromSourceWhileExiledTriggeredAbility.java]: Done

**Pull request merged**